### PR TITLE
Refactor network client

### PR DESCRIPTION
**Pull request by bowenwang1996**
Addresses https://github.com/nearprotocol/nearcore/issues/107.

### CODE REVIEW

File: node/network/src/protocol.rs [hunk 1]
```diff
@@ -73,7 +73,7 @@ pub struct Protocol<B: Block, H: ProtocolHandler> {
     // info about peers
     peer_info: RwLock<HashMap<NodeIndex, PeerInfo>>,
     // backend client
-    client: Arc<Client<B>>,
+    client: Arc<Chain<B>>,
```
Comment (azban) [inline node/network/src/protocol.rs]: should be chain

File: node/network/src/service.rs [hunk 2]
```diff
@@ -29,7 +29,7 @@ impl<B: Block, H: ProtocolHandler> Service<B, H> {
         config: ProtocolConfig,
         net_config: NetworkConfiguration,
         handler: H,
-        client: Arc<Client<B>>,
+        client: Arc<Chain<B>>,
```
Comment (azban) [inline node/network/src/service.rs]: should be chain

**Comment by azban** [comment]
to clarify, some of these formats are legit, but mostly the ones where we move away from splitting on multiple lines don't look great..

File: node/network/src/protocol.rs (hunk 1 above)
Comment (bowenwang1996) [reply node/network/src/protocol.rs]: Fixed in 721b876

File: node/network/src/service.rs (hunk 2 above)
Comment (bowenwang1996) [reply node/network/src/service.rs]: Fixed in 721b876

**Comment by bowenwang1996** [comment]
Fixed some fmt formatting in a1c1cbd

**Pull request merged**